### PR TITLE
feat: add docs 'sending only necessary data to client'

### PR DESCRIPTION
**Pull request by devartes**
- I didn't include images because I noticed they are hosted on Amazon S3, and I don't have access to buckets if that's the case. I would appreciate suggestions for code.

### CODE REVIEW

**Review by lucis — CHANGES_REQUESTED** [review]

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+description: TODO
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: Done

Captura de Tela 2023-12-05 às 18 23 38

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+since: 1.0.0
+---
+
+Em algumas situações, ao utilizar [loaders](http://localhost:8000/docs/pt/concepts/loader) para obter dados, pode-se receber uma quantidade excessiva de informações, impactando negativamente o desempenho da página. Isso é evidenciado pelo tamanho significativo do JSON transmitido para o cliente.
+
+![287813945-9ccd40a6-f41f-486f-a2b6-3f1efca0bfd1](https://github.com/deco-sites/starting/assets/76822093/bff19a56-bc1d-475b-b25d-f45fd1af713a)
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: Done

Captura de Tela 2023-12-05 às 18 26 08

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
+---
+description: TODO
+since: 1.0.0
+---
+
+Em algumas situações, ao utilizar [loaders](http://localhost:8000/docs/pt/concepts/loader) para obter dados, pode-se receber uma quantidade excessiva de informações, impactando negativamente o desempenho da página. Isso é evidenciado pelo tamanho significativo do JSON transmitido para o cliente.
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: Done

Captura de Tela 2023-12-05 às 18 29 33
Captura de Tela 2023-12-05 às 18 28 50

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+![287813945-9ccd40a6-f41f-486f-a2b6-3f1efca0bfd1](https://github.com/deco-sites/starting/assets/76822093/bff19a56-bc1d-475b-b25d-f45fd1af713a)
+
+
+Para mitigar esse problema, propomos a implementação de um processo de pré-processamento dos dados no lado do cliente, garantindo que apenas as informações necessárias sejam transmitidas e utilizadas no markup/UI.
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: Done

Captura de Tela 2023-12-05 às 18 31 33

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+## Fluxo de Dados
+
+1. Solicitação de Dados: Inicie o fluxo normal de dados, solicitando as informações necessárias por meio de props.
+
+2. Loader de Dados: Utilize um loader para obter os dados desejados. Em algumas situações, o loader pode retornar uma quantidade substancial de dados, por exemplo, ao solicitar produtos relacionados da [VTEX](https://www.deco.cx/docs/pt/composable-apis/vtex). No entanto, é importante observar que há momentos em que o loader pode retornar uma quantidade de dados maior do que o necessário para a operação em questão.
+
+3. Pré-Processamento Inline: Introduza um mecanismo de pré-processamento de dados inline. Este componente receberá as mesmas props do loader e processará os dados de forma a transmitir apenas as informações essenciais, otimizando assim o desempenho.
+
+4. Entrega ao Componente: Transmita apenas os dados processados para o componente JSX, reduzindo assim a carga desnecessária no cliente.
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: Is it better like this?

Captura de Tela 2023-12-05 às 18 49 10
Captura de Tela 2023-12-05 às 18 51 07

File: docs/performance/sending-only-necessary-data-to-client/pt.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  layout: Layout;
+}
+
+/** @title Product Card Layout */
+const loader = ({ layout }: Props): Layout => layout;
+
+export const Preview = (props: Props) => {
+  const { layout } = props;
+
+  return (
```
[the remark this answers is not in the document]
Comment (devartes) [reply docs/performance/sending-only-necessary-data-to-client/pt.md]: I included the code for this loader as an example, but I have doubts if it's okay like this

Captura de Tela 2023-12-05 às 18 53 35

**Pull request merged**